testing/test-helper.js

```diff
@@ -105,6 +105,52 @@ export function assertScreenReaderElement(element) {
   expect(computedStyle.getPropertyValue('visibility')).to.equal('visible');
 }
 
+/**
+ * Expects the function call to generate console error messages.
+ *
+ * The expectedErrorMessages parameter is a list of strings or RegExp objects
+ * to match against the output of each call to console.error.
+ *
+ * e.g.:
+ * expectConsoleError(() => {
+ *   console.error('This is the first error');
+ *   console.warn('Warns and logs are not caught');
+ *   console.error('This is the second error');
+ * }, ['This is the first error', /second/]);
+ *
+ * @param {!Function} func
+ * @param {!Array<string, !RegExp>} expectedErrorMessages
+ */
+export function expectConsoleError(func, expectedErrorMessages) {
+  let callIndex = 0;
+  console.error.callsFake((...messages) => {
+    // On every call to console.error assert that there are still expected
+    // error messages left, and assert that the message matches the one
+    // that was expected for this call.
+    if (callIndex >= expectedErrorMessages.length) {
```

Comment:
This logic seems brittle. What if the errors are logged in non-deterministic order, or if a particular error is logged more than once, or not at all?

I think we should simply take a collection `expectedErrorMessages`, and throw if an error that's not in the collection happens to be logged.

Reply:
It's fair to expect the error messages in a particular order. Most testing primitives act this way - we can add an "unordered" version later if there's demand for it